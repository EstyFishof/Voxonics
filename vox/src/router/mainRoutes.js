import agentPanel from "@/assets/img/icons/agent-panel.svg";
import reports from "@/assets/img/icons/reports.svg";
import settings from "@/assets/img/icons/setting.svg";
import dashboard from "@/assets/img/icons/dashboard.svg";
import support from "@/assets/img/icons/support.svg";
import dialer from "@/assets/img/icons/dialer.svg";
export const routes = [
    {
        path: "/",
        component: () => import("@/views/Login.vue"),
        meta: { layout: "login" },
    },
    {
        path: "/agent-panel",
        component: () => import("@/views/AgentPanel.vue"),
        meta: { title: "Agent Panel", icon: agentPanel },
    },
    {
        path: "/phone",
        component: () => import("@/views/Phone/Phone.vue"),
        meta: { title: "Soft Phone", icon: agentPanel },
    },
    {
        path: "/dialer",
        component: () => import("@/views/Dialer/Dialer.vue"),
        redirect: "dialer/agents-table",
        children: [
            {
                path: "agents-table",
                component: () => import("@/views/Dialer/AgentsTable.vue"),
                meta: {
                    title: "Dialer",
                    icon: dialer,
                },
            },
        ],
    },
    {
        path: "/reports",
        component: () => import("@/views/Reports/Reports.vue"),
        redirect: "/reports/call-history",
        children: [
            {
                path: "call-history",
                component: () => import("@/views/Reports/CDR/CDR.vue"),
                meta: { title: "Reports", icon: reports },
            },
            {
                path: "billing/charges",
                component: () => import("@/views/Reports/BillingReports/Charges.vue"),
                meta: { title: "Reports", icon: reports },
            },
            {
                path: "billing",
                component: () => import("@/views/Reports/BillingReports/Statement.vue"),
                meta: { title: "Reports", icon: reports },
            },
            {
                path: "billing/payments",
                component: () => import("@/views/Reports/BillingReports/Payments.vue"),
                meta: { title: "Reports", icon: reports },
            },
        ],
    },
    {
        path: "/config",
        component: () => import("@/views/Config/Config.vue"),
        redirect: "/config/users",
        children: [
            {
                path: "users",
                component: () => import("@/views/Config/Users/Users.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "users/create-user",
                component: () => import("@/views/Config/Users/CreateUser/CreateUser.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "users/account-info/:id",
                component: () => import("@/views/Config/Users/Accountinfo/Accountinfo.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "users/edit-user/:id",
                component: () => import("@/views/Config/Users/EditUser/EditUser.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "departments",
                component: () => import("@/views/Config/Departments/Departments.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "departments/create-department",
                component: () => import("@/views/Config/Departments/CreateDepartment/CreateDepartment.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "departments/department-setting/:id",
                component: () => import("@/views/Config/Departments/DepartmentSettings/DepartmentSettings.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "external-numbers",
                component: () => import("@/views/Config/Numbers/WrapNumbers"),
                redirect: "external-numbers/info",
                children: [
                    {
                        path: "info",
                        component: () => import("@/views/Config/Numbers/Numbers.vue"),
                        meta: { title: "Settings", icon: settings },
                    },
                    {
                        path: "departments",
                        component: () => import("@/views/Config/Numbers/NumbersOfDepartments.vue"),
                        meta: { title: "Settings", icon: settings },
                    },
                ],
            },

            {
                path: "queues",
                component: () => import("@/views/Config/Queues/Queues.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "queues/create-queue",
                component: () => import("@/views/Config/Queues/CreateQueue/CreateQueue.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "queues/queue-setting/:id",
                component: () => import("@/views/Config/Queues/QueuesSettings/QueuesSettings.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "incoming-route",
                component: () => import("@/views/Config/Scenarios/ParentScenario.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "whitelistIP",
                component: () => import("@/views/Config/WhitelistIP/WhitelistIP.vue"),
                meta: { title: "Settings", icon: settings },
            },
            {
                path: "outboundCalls",
                component: () => import("@/views/Config/OutboundCalls/OutboundCalls.vue"),
                meta: { title: "Settings", icon: settings },
            },
        ],
    },
    {
        path: "/dashboard",
        component: () => import("@/views/Dashboard/Dashboard.vue"),
        redirect: "dashboard/agents-map",
        children: [
            {
                path: "agents-map",
                component: () => import("@/views/Dashboard/AgentsMap.vue"),
                meta: { title: "Dashboard", icon: dashboard },
            },
        ],
    },
    {
        path: "/support",
        component: () => import("@/views/Support/Manual.vue"),
        redirect: "/support/manual",
        children: [
            {
                path: "manual",
                component: () => import("@/views/Support/Manual.vue"),
                meta: { title: "Support", icon: support },
            },
        ],
    },
];

