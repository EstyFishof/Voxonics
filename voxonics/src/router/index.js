import { createRouter, createWebHistory } from 'vue-router';
import HelloWorld from '../components/HelloWorld.vue';
import HomePage from '../components/HomePage.vue';

const routes = [
  {
    path: '/',
    name: 'HomePage',
    component: HomePage
  },
  {
    path: '/helloworld/:msg',
    name: 'HelloWorld',
    component: HelloWorld,
    props: true // Pass route params as props
  }
];

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
});

export default router;
