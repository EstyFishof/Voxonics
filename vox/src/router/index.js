import Vue from 'vue'
import VueRouter from 'vue-router'
import store from '../store/index'
import {routes} from '@/router/mainRoutes'


Vue.use(VueRouter)


const router = new VueRouter({
    mode: 'history',
    base: '/',
    routes
})

routes.forEach(route => {
    router.addRoute(route)
})


router.beforeEach((to, from, next) => {
    if (from.path !== '/') {
        for (const abortController of store.getters.abortControllerPool) {
            abortController.abort()
            store.commit('removeAbortController', abortController)
            store.dispatch('setSpinner', false)
        }
    }
    if (to.path.split('/')?.[1] === 'admin') {
        store.commit('setCurrentRoutes', 'admin')
    } else {
        store.commit('setCurrentRoutes', 'client')
    }
    store.commit('setMeta', to.meta)
    if (to.path !== '/') {
        if (!store.getters.userInfo?.name) {
            store.dispatch('getUserInfo')
        }
    }
    if (to.path === '/' && localStorage.getItem('token')) {
        return next('/agent-panel')
    }

    if (localStorage.getItem('role') === 'agent' || localStorage.getItem('role') === undefined || localStorage.getItem('role') === null) {
        if (to.path !== '/agent-panel' && to.path !== '/phone' && to.path !== `/config/users/account-info/${store.getters.userInfo?.internalNumber}` && to.path !== '/') {
            return next('/agent-panel')
        }
    }

    if (to.redirectedFrom) {
        if (to.redirectedFrom === '/config' && from.path.indexOf('config') !== -1) {
            return next(false)
        }
    }
    let indexConfig = to.path.indexOf('config')
    let indexReports = to.path.indexOf('reports')
    if (indexConfig === 1) {
        store.dispatch('getIsPage', 'config')
        let arr = to.path.slice(indexConfig + 7).split('/')
        store.dispatch('getPath', arr)
        next()
    } else if (indexReports === 1) {
        store.dispatch('getIsPage', 'reports')
        let arr = to.path.slice(indexReports + 8).split('/')
        store.dispatch('getPath', arr)
        next()
    } else {
        store.dispatch('getIsPage', '')
        let arr = to.path.split('/')
        store.dispatch('getPath', arr)
    }
    next()
})
export default router
