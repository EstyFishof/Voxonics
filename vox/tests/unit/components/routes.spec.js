import { routes } from '@/router';

describe('routes', () => {
    it('should be array', () => {
        expect(routes.length).toBeGreaterThan(0);
    });
    it('routes snapshot', () => {
        expect(routes).toMatchSnapshot();
    });
    test.each(routes)(`should have a component %s`, (route) => {
        expect(route).toHaveProperty('path');
        expect(route).toHaveProperty('name');
        expect(route).toHaveProperty('component');
        const instance = route.component();
        expect(instance).toBeTruthy();
        expect(route.component).toBeDefined();
    });
    test.each(routes[2].children)(`should have a component children`, (child) => {
        expect(child).toHaveProperty('path');
        expect(child).toHaveProperty('name');
        expect(child).toHaveProperty('component');
        const instance = child.component();
        expect(instance).toBeTruthy();
        expect(child.component).toBeDefined();
    });
});
