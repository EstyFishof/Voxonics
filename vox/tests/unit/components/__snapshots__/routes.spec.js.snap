// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes routes snapshot 1`] = `
Array [
  Object {
    "component": [Function],
    "meta": Object {
      "layout": "login",
    },
    "name": "Login",
    "path": "/",
  },
  Object {
    "component": [Function],
    "meta": Object {
      "icon": "",
      "title": "Agent Panel",
    },
    "name": "AgentPanel",
    "path": "/agent-panel",
  },
  Object {
    "children": Array [
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Reports",
        },
        "name": "CDR",
        "path": "call-history",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Reports",
        },
        "name": "Charges",
        "path": "billing/charges",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Reports",
        },
        "name": "Statement",
        "path": "billing",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Reports",
        },
        "name": "Payments",
        "path": "billing/payments",
      },
    ],
    "component": [Function],
    "name": "Reports",
    "path": "/reports",
    "redirect": "/reports/call-history",
  },
  Object {
    "children": Array [
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Settings",
        },
        "name": "Users",
        "path": "users",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Settings",
        },
        "name": "CreateUser",
        "path": "users/create-user",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Settings",
        },
        "name": "AccountInfo",
        "path": "users/account-info/:id",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Settings",
        },
        "name": "EditUser",
        "path": "users/edit-user/:id",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Settings",
        },
        "name": "Departments",
        "path": "departments",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Settings",
        },
        "name": "CreateDepartment",
        "path": "departments/create-department",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Settings",
        },
        "name": "DepartmentSettings",
        "path": "departments/department-setting/:id",
      },
      Object {
        "children": Array [
          Object {
            "component": [Function],
            "meta": Object {
              "icon": "",
              "title": "Settings",
            },
            "name": "Numbers",
            "path": "info",
          },
          Object {
            "component": [Function],
            "meta": Object {
              "icon": "",
              "title": "Settings",
            },
            "name": "NumbersOfDepartments",
            "path": "departments",
          },
        ],
        "component": [Function],
        "name": "WrapNumbers",
        "path": "external-numbers",
        "redirect": "external-numbers/info",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Settings",
        },
        "name": "Queues",
        "path": "queues",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Settings",
        },
        "name": "CreateQueue",
        "path": "queues/create-queue",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Settings",
        },
        "name": "QueueSetting",
        "path": "queues/queue-setting/:id",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Incoming Routing",
        },
        "name": "ParentScenario",
        "path": "incoming-route",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Settings",
        },
        "name": "whitelistIP",
        "path": "whitelistIP",
      },
    ],
    "component": [Function],
    "name": "Config",
    "path": "/config",
    "redirect": "/config/users",
  },
  Object {
    "children": Array [
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Dashboard",
        },
        "name": "AgentsMap",
        "path": "agents-map",
      },
    ],
    "component": [Function],
    "name": "Dashboard",
    "path": "/dashboard",
    "redirect": "dashboard/agents-map",
  },
  Object {
    "component": [Function],
    "name": "Dialer",
    "path": "/dialer",
  },
  Object {
    "children": Array [
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Support",
        },
        "name": "Manual",
        "path": "manual",
      },
      Object {
        "component": [Function],
        "meta": Object {
          "icon": "",
          "title": "Support",
        },
        "name": "ContactUs",
        "path": "contact-us",
      },
    ],
    "component": [Function],
    "name": "Support",
    "path": "/support",
    "redirect": "/support/manual",
  },
]
`;
